### ISSUE TITLE
How to tune timeouts/wait times

### ISSUE THREAD

**Issue by v173k** [felis/USB_Host_Shield_2.0#116]
Hello,

I am using the USB Host Shield with a gps module which is sending constant serial data via a PL2303 USB to serial converter chip.

I am facing what appears to be buffer overflows problems. Everytime the gps module sends more then the usual amount of data I start getting stretches of missing data from the stream. The baud rate is set to 115200 and should be sufficient for the amount of data being sent. In the past when working with Serial data for a similar application (but without using a USB shield) I was able to solve this problem by reducing the Serial wait time using Serial.setTimeout() so I was wondering if there was a similar setting in the USB_Host_Shield library.

I notice there are some settings for changing NAK handling ( `bmNakPower` ) and specifically some settings in cdcacm.cpp:

    //epInfo[i].bmNakPower = USB_NAK_NOWAIT;
    epInfo[i].bmNakPower = USB_NAK_MAX_POWER;

However, I wasn't sure what these settings where for exactly or if they would help my buffer overflow problems.

So in summary I was hoping you could point me towards what settings could be modified to tune the timeouts and explain what the bmNakPower variable was used for. I was also wondering how an interrupt function similar to serialEvent() could be implemented for the USB shield.

Thank you
Vivek

**Comment by Lauszus** [comment]
I believe this issue was solved, so I'm closing it. Please reopen if necessary.
